software/deploy/deploy_test.go

```diff
@@ -452,9 +452,84 @@ func (s *Suite) TestAirflowSuccessForImageOnly() {
 	houstonMock.On("GetRuntimeReleases", "").Return(mockRuntimeReleases, nil)
 
-	_, err := Airflow(houstonMock, "./testfiles/", "test-deployment-id", "test-workspace-id", "", false, false, false, description, true)
+	_, err := Airflow(houstonMock, "./testfiles/", "test-deployment-id", "test-workspace-id", "", false, false, false, description, true, "")
 	s.NoError(err)
 	houstonMock.AssertExpectations(s.T())
+	mockImageHandler.AssertExpectations(s.T())
+}
+
+func (s *Suite) TestAirflowSuccessForImageName() {
+	fs := afero.NewMemMapFs()
+	configYaml := testUtil.NewTestConfig("localhost")
+	afero.WriteFile(fs, config.HomeConfigFile, configYaml, 0o777)
+	config.InitConfig(fs)
+
+	customImageName := "test-image-name"
+	mockImageHandler := new(mocks.ImageHandler)
+	imageHandlerInit = func(image string) airflow.ImageHandler {
+		mockImageHandler.On("Push", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil)
+		mockImageHandler.On("TagLocalImage", customImageName).Return(nil)
+		mockImageHandler.On("GetLabel", "", airflow.RuntimeImageLabel).Return("test", nil)
+		return mockImageHandler
+	}
+
+	mockedDeploymentConfig := &houston.DeploymentConfig{
+		AirflowImages: mockAirflowImageList,
+	}
+	mockRuntimeReleases := houston.RuntimeReleases{
+		houston.RuntimeRelease{Version: "4.2.4", AirflowVersion: "2.2.5"},
+		houston.RuntimeRelease{Version: "4.2.5", AirflowVersion: "2.2.5"},
+	}
+	houstonMock := new(houston_mocks.ClientInterface)
+	houstonMock.On("GetWorkspace", mock.Anything).Return(&houston.Workspace{}, nil).Once()
+	houstonMock.On("ListDeployments", mock.Anything).Return([]houston.Deployment{{ID: "test-deployment-id"}}, nil).Once()
+	houstonMock.On("GetDeploymentConfig", nil).Return(mockedDeploymentConfig, nil).Once()
+	dagDeployment := &houston.DagDeploymentConfig{
+		Type: "dag-only",
+	}
+	deployment := &houston.Deployment{
+		DagDeployment: *dagDeployment,
+	}
+
+	houstonMock.On("GetDeployment", mock.Anything).Return(deployment, nil).Once()
+	houstonMock.On("GetRuntimeReleases", "").Return(mockRuntimeReleases, nil)
+
+	_, err := Airflow(houstonMock, "./testfiles/", "test-deployment-id", "test-workspace-id", "", false, false, false, description, true, customImageName)
+	s.NoError(err)
+	houstonMock.AssertExpectations(s.T())
+	mockImageHandler.AssertExpectations(s.T())
+}
+
+func (s *Suite) TestAirflowFailForImageNameWhenImageHasNoRuntimeLabel() {
+	fs := afero.NewMemMapFs()
+	configYaml := testUtil.NewTestConfig("localhost")
+	afero.WriteFile(fs, config.HomeConfigFile, configYaml, 0o777)
+	config.InitConfig(fs)
+
+	customImageName := "test-image-name"
+	mockImageHandler := new(mocks.ImageHandler)
+	imageHandlerInit = func(image string) airflow.ImageHandler {
+		mockImageHandler.On("TagLocalImage", customImageName).Return(nil)
+		mockImageHandler.On("GetLabel", "", airflow.RuntimeImageLabel).Return("", nil)
+		return mockImageHandler
+	}
+
+	houstonMock := new(houston_mocks.ClientInterface)
```

Comment:
nit: similar to the other PR, can we start moving this setup and teardown steps common between these tests to the suite's setup and teardown function? That is one of the reasons we introduced a suite to CLI's unit test in order to simplify some of these tests.